### DISCUSSION TITLE
Before(Test) not working as expected

### DISCUSSION THREAD

**Discussion by mohaaron** [thomhurst/TUnit#1041] [General]
I have started trying to use TUnit in a sandbox project I've been testing ideas with and am finding that the attribute Before(Test) is not doing what I would have expected. It should be run before each test but it's been running in a sequence ahead of all the tests. Meaning that if I have three tests the Before method runs three times in a row and then runs the three tests afterwards. I was hoping I could use the Before(Test) to reset the data being used in each test before the test is run. How would I reset data before each test is run?

TUnit Version: 0.2.55
Demo: https://github.com/mohaaron/BlazingBudget/tree/trunk/tests/BlazingBudget.TUnitTesting.Tests

**Comment by thomhurst** [comment]
TUnit runs in parallel by default, and you get a new class instance for each test. If your data isn't static, then you don't need to reset as your objects will be isolated to just the running test

**Reply by thomhurst** [reply]
In your example, you could remove the before method entirely and the tests would all pass, because each test has a `new TUnitTests()`. And so your field initializer `private int standardValue = 1;` would be called for each test.

**Reply by mohaaron** [reply]
@thomhurst, thank you for the fast response.

I understand and realize that I'm also trying to do this in another project that I can't share using an in-memory Sqlite db that's having issues where the domain entity being tested is generating a strongly typed id sequence that is not what I expect and one of the tests is always failing. I think I have written my in-memory db in a way that just isn't going to work. I have so far found that if I use the static Before(Class) method my tests all pass. This is due to the DbContext being created only once so I have a consistent set of data for all the tests.

**Reply by thomhurst** [reply]
If you wanted 

> Before > Test1 > Before > Test2 > Before > Test3

Then you'd have to turn off parallelism with the `[NotInParallel]` attribute
